### PR TITLE
ES6 comment count

### PR DESCRIPTION
**Pull request by regiskuckaertz**
For some reason, the `raw-loader` returns empty `DIV`s in my tests. I hereby summon the wizards of Jest @gustavpursche and @SiAdcock to help me figure that one out.

### CODE REVIEW

**Comment by SiAdcock** [comment]
To minimise inconvenience, all [templates are mocked](https://github.com/guardian/frontend/blob/master/__mocks__/templateMock.js) by default

You can get around this by mocking the templates explicitly in the test spec:
```js
jest.mock(
    'raw-loader!common/views/discussion/comment-count.html',
    () => '<a href="boop"></a>'
);
```

**Comment by regiskuckaertz** [comment]
Sir, you saved my day. Thanks for the info!

**Comment by SiAdcock** [comment]
😄 

It's what I do 🙇 

picture 270

**Comment by regiskuckaertz** [comment]
says he before disappearing in the horizon

giphy

File: static/src/javascripts/projects/common/modules/discussion/comment-count.spec.js
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/discussion/comment-count.spec.js]: Since jest is based on jsdom, I'd rather avoid adding complexity with fastdom and ditch it.

File: static/src/javascripts/projects/common/modules/discussion/comment-count.js
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/discussion/comment-count.js]: Yep, good call!

File: static/src/javascripts/projects/common/modules/discussion/comment-count.js
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/comment-count.js]: Let's give this function a return type

File: static/src/javascripts/projects/common/modules/discussion/comment-count.js
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/comment-count.js]: Another missing return type here

**Comment by SiAdcock** [comment]
Amazing conversion, man, a thing of beauty 🌆  👨‍🎨

File: static/src/javascripts/projects/common/modules/discussion/comment-count.js
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/discussion/comment-count.js]: That's typically the kind of function you'd want to put in a library

**Comment by regiskuckaertz** [comment]
Is there someone in who could approve this PR, probably from discussion?

**Review by SiAdcock — APPROVED** [review]

**Comment by SiAdcock** [comment]
Looks good, dude. You're unlikely to find someone who understands this code intimately. Next week we're doing some work to make it easier to test discussion locally, so you may want to wait until then to merge. We have introduced a number of bugs into discussion as part of the ES6 conversion process.

**Comment by regiskuckaertz** [comment]
Sounds good to me. I saw the next on my list is `comments` which is using the infamous `Component`. 😨 

...

*moves comments module to Simon's list*

**Comment by gustavpursche** [comment]
@regiskuckaertz It's already converted, but broke everything. As soon as comments are running locally, I'll move this horror away from you :)

**Comment by regiskuckaertz** [comment]
@gustavpursche thanks dude 😂  good luck!

**Comment by regiskuckaertz** [comment]
Closing this because of 😨

**Pull request closed without merging**
